Contributor Onboarding (First chapter or two) (#12)
tomeshnet/toronto-community-network-docs: src/onboarding/github.md

# GitHub Tickets

GitHub is a service that provides code hosting and versioning. We also use it for its collaborative features, such as issue/bug tracking and project boards. This documents explains how to use GitHub to keep track of tasks.

## Repositories

> Here is the homepage for the [toronto-community-network](https://github.com/tomeshnet/toronto-community-network) repository.

Reviewer: The text is a little akward
`Here is the homepage for the toronto-community-network repository.`
Unless you include the link the language does not make sense. 

Also is it a homepage?

Maybe

```suggestion
The main repository for the Toronto Community Network can be found on github under [tomeshenet/toronto-community-network](https://github.com/tomeshnet/toronto-community-network).
```

![homepage](./images/homepage.png)

> A repository (or "repo") is like a project on GitHub, with a collection of files and a name. You can see the name of this one, `toronto-community-network` and that it was created by the [tomeshnet](https://github.com/tomeshnet) organization.

Reviewer: Maybe "project" would be confused with "github project". `like a project` does not really add anything as project is not well defined.  Also refering to it "on GitHub" adds to the confusion vs "github projects".

Also not sure about the tone. If this is a manual it should not use "you can see" like wording.

```suggestion
A repository (or "repo") is a collection of files, issues, and project boards grouped uner a name. The name of this one is `toronto-community-network` and that it was created by the [tomeshnet](https://github.com/tomeshnet) organization.
```

> You can see an overview of the files and folders in the middle of screen, and below that you can see what's called the README, which is a file that explains what the repository is for. If you click on `README.md` you'll see it has the same contents as the text on the homepage.

Reviewer: Again not sure about the tone

"You can see an overview of the files and folders" vs "An overview of the files and folders is displayed in the middle of the screen"

> You can see an overview of the files and folders in the middle of screen, and below that you can see what's called the README, which is a file that explains what the repository is for. If you click on `README.md` you'll see it has the same contents as the text on the homepage.

Reviewer: ```suggestion
You can see an overview of the files and folders in the middle of the screen, and below that you can see what's called the README, which is a file that explains what the repository is for. If you click on `README.md` you'll see it has the same contents as the text on the homepage.
```

## Issues

> If you click on `Issues` tab you'll see this:

Reviewer: "you" Tone 
In fact, this line may be not needed at all.

```suggestion
```

![issues](./images/issues.png)

> This tab provides a list of all the issues that have been created for this repository. Issues are similar to the idea of "tickets" or bug reports. It allows us to keep track of tasks or potential plans. All issues have numbers, referenced with a hashtag, such as #32, which can be found [here](https://github.com/tomeshnet/toronto-community-network/issues/32).

Reviewer: "allows us" tone

Also hashtag notation - actually gitHub Autolinked References
https://docs.github.com/en/github/writing-on-github/autolinked-references-and-urls

> For any issue, there is an initial comment explaining what the issue is for, and a discussion going on below about the issue, with reactions, editing, links, etc. The initial comment will change and get edited by others over time as new information is known.

Reviewer: - There is a template that can be used for tasks
- We may add other templates over time (such as request for expense repayment)

Reviewer 2: Added in a2fa812, let me know if I'm missing anything.

Reviewer: good enough. Thanks

Toronto Mesh uses issue templates, so some text is filled out already when a user opens an issue. At time of writing there is only one template, used for tasks, which can be seen if you [open an issue](https://github.com/tomeshnet/toronto-community-network/issues/new?template=task_template.md). More templates may be added in the future.

Issues also have assignees, which can be seen in the sidebar. These are the users responsible for completing this task. A user can assign themselves or other people.

> Issues can either be open or closed, which denotes their completion. Issues are closed once they're no longer relevant, or when the original task or bug has been fixed.

Reviewer: ```suggestion
Issues can either be open or closed, which denotes their completion. Issues are closed once they are no longer relevant, or when the original task or bug has been fixed.
```

## Labels

Issues can have labels, projects, and milestones, which are shown on the sidebar. If a label is clicked, all the issues with that label can be viewed. For example, here are all the communications issues:

![communications issues](./images/issue_comms_label.png)

> Using the sidebar, you can filter by other labels if needed.

Reviewer: ```suggestion
On the sidebar there is a filter for labels if needed.
```

![filter labels modal](./images/filter_labels.png)

## Projects

[#32](https://github.com/tomeshnet/toronto-community-network/issues/32) is also linked to a project, as can be seen in the sidebar.

![sidebar project](./images/sidebar_project.png)

> If you click that text or click the Projects tab at the top, you'll be able to see the project. Repositories can have multiple projects, but in this repository there is just one. Here is what it looks like:

Reviewer: "if you" tone

![project view](./images/project.png)

A GitHub Project is a type of [Kanban](https://en.wikipedia.org/wiki/Kanban_%28development%29) board for organizing issues.
Each of the cards are issues, which can be in different columns. We use a standard setup with To Do, Doing (aka "In Progress"), and Completed columns.

> We also have a Backlog column on the far left for low-priority tasks. Issues can be moved from column to column as needed, to indicate their status. Issues will automatically be moved to Completed when they're closed.

Reviewer: ```suggestion
We also have a Backlog column on the far left for low-priority tasks. Issues can be moved from column to column as needed, to indicate their status. Issues will automatically be moved to Completed when they are closed.
```
